include/zephyr/shell/shell.h

```diff
@@ -952,7 +952,7 @@ extern void z_shell_print_stream(const void *user_ctx, const char *data,
 	static struct shell_ctx UTIL_CAT(_name, _ctx);                                             \
 	Z_SHELL_HISTORY_DEFINE(_name##_history, CONFIG_SHELL_HISTORY_BUFFER);                      \
 	Z_SHELL_FPRINTF_DEFINE(_name##_fprintf, &_name, _out_buf, CONFIG_SHELL_PRINTF_BUFF_SIZE,   \
-			       true, z_shell_print_stream);                                        \
+			       CONFIG_SHELL_PRINTF_AUTOFLUSH, z_shell_print_stream);               \
```

Comment:
`CONFIG_SHELL_PRINTF_AUTOFLUSH` would expand to nothing if not enabled.

```suggestion
			       IS_ENABLED(CONFIG_SHELL_PRINTF_AUTOFLUSH), z_shell_print_stream);   \
```

Reply:
Added, thank you